lib/teleterm/apiserver/grpccredentials.go

```diff
@@ -65,22 +65,32 @@ func createServerCredentials(serverKeyPair tls.Certificate, clientCertPath strin
 }
 
 func generateAndSaveCert(targetPath string) (tls.Certificate, error) {
-	// File is first saved using under tempPath and then renamed to targetPath.
-	// This prevents other processes from reading a half written file.
-	tempPath := targetPath + ".tmp"
+	// The cert is first saved under a temp path and then renamed to targetPath. This prevents other
+	// processes from reading a half-written file.
+	tempFile, err := os.CreateTemp(filepath.Dir(targetPath), filepath.Base(targetPath))
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	defer os.Remove(tempFile.Name())
 
 	cert, err := utils.GenerateSelfSignedCert([]string{"localhost"})
 	if err != nil {
 		return tls.Certificate{}, trace.Wrap(err, "failed to generate the certificate")
 	}
 
-	err = os.WriteFile(tempPath, cert.Cert, 0600)
-	if err != nil {
+	if err = tempFile.Chmod(0600); err != nil {
```

Comment:
I'm not sure if I need that extra `Chmod` here. AFAIK `os.CreateTemp` already creates a file with 0600 and I suppose we can expect a temp file to have the strictest permissions possible. But the docs for `os.CreateTemp` don't state any guarantees about it so I'd rather be explicit.